### PR TITLE
#1282 git misspelled

### PR DESCRIPTION
**Pull request by NikosKakonas**
The function git_support can not be useful if a command has the word "git" misspelled.
The purpose of this rule is to unfuck the commands like: gti commit -m 'message'.
Without this rule, for example, if you try to unfuck this command the result will be:
git commit -m message. This means that the quotation marks do not return. #1282

This rule is generally useful for every git command, in which the word git is misspelled.

I have also run some tests!

### CODE REVIEW

File: tests/rules/test_git_misspenlled.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+@pytest.mark.parametrize(
+    "script, output, new_command",
+    [
+        ('''gti commit -m "this is a message"''', "", '''git commit -m "this is a message"'''),
+        ('''tgi commit -m "this is a message"''', "", '''git commit -m "this is a message"'''),
+    ],
+)
+def test_get_new_command(script, output, new_command):
+    assert get_new_command(Command(script, output)) == new_command
```
Comment (scorphus) [inline tests/rules/test_git_misspenlled.py:26]: Please, correct the name of this file.

Also, could you add `test_not_match` for cases where the rule does not match?

**Review by scorphus — APPROVED** [review]
This could be helpful for cases `no_command` isn't able to fix. Please, consider my suggestions below.

**Comment by NikosKakonas** [comment]
I made test_not_match fucntion. I hope that now it is better!

**Comment by djh82** [comment]
It feels like rather than solving this specifically for the git command, it might be prudent to solve the issue in the no_command rule?

**Comment by NikosKakonas** [comment]
Thank you a lot for the review! Is this commit satisfying for the project?

**Comment by djh82** [comment]
You'll probably want to revert the readme change and add some further tests for no_command?

**Comment by scorphus** [comment]
Regarding the failure, please check #1310.

**Comment by NikosKakonas** [comment]
@djh82 @scorphus
Are there any other mistakes of missing cases?

**Comment by NikosKakonas** [comment]
I believe that now it is correct!

**Review by scorphus — CHANGES_REQUESTED** [review]

**Comment by scorphus** [comment]
@NikosKakonas, could you please rebase this on top of current `master`? Thanks!

**Comment by NikosKakonas** [comment]
> @NikosKakonas, could you please rebase this on top of current `master`? Thanks!

image

@scorphus Could you please help me with this message?

**Comment by NikosKakonas** [comment]
@scorphus  Is it now okay? Thanks!

**Comment by NikosKakonas** [comment]
@scorphus Is it okay for merging?

**Review by scorphus — APPROVED** [review]
Thanks, @NikosKakonas! 👍

**Comment by NikosKakonas** [comment]
The plesure is main!

**Comment by alaycock** [comment]
@nkakonas does this justify a new release? This has been bugging me for a bit.

**Comment by alaycock** [comment]
@scorphus could you cut a new release for this?

**Pull request merged**